src/components/UndoRedoButtons.tsx

```diff
@@ -20,15 +21,15 @@ export function UndoRedoButtons({
         iconName="arrowTurnLeft"
         onClick={() => kclManager.undo()}
         className="rounded-r-none"
-        disabled={undoDepth(kclManager.editorState) === 0}
+        disabled={kclManager.undoDepth.value === 0}
```

Comment:
This reactivity implicitly relied on `editorState` being created and destroyed all the time, so I had to update it to be a signal inside of `kclManager` instead.